hypothesis-python/src/hypothesis/internal/filtering.py

```diff
@@ -73,6 +73,10 @@ def convert(node: ast.AST, argname: str) -> object:
         if node.id != argname:
             raise ValueError("Non-local variable")
         return ARG
+    if isinstance(node, ast.Call):
+        # if node.func.id == "len":
+        if isinstance(node.func, ast.Name) and node.func.id == "len":
+            return ARG
```

Comment:
```suggestion
        if isinstance(node.func, ast.Name) and node.func.id == "len" and len(node.args) == 1:
            convert(node.args[0], argname)  # comparison must be to len *of the lambda arg*
            return ARG
```

Reply:
Do you purposefully leave the `return ARG` here to indicate that we're returning `ARG`, even though it would do the same thing as `convert(...)`?